flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -213,10 +226,11 @@ public static void main(String[] args) throws Exception {
 			LOG.info("Cannot determine the maximum number of open file descriptors");
 		}
 
-		ParameterTool parameterTool = ParameterTool.fromArgs(args);
-
-		final String configDir = parameterTool.get("configDir");
+		// try to parse the command line arguments
+		CommandLineParser parser = new DefaultParser();
+		CommandLine cmd = parser.parse(ALL_OPTIONS, args);
 
+		final String configDir = cmd.getOptionValue("configDir");
```

Comment:
Have you tried whether this actually works? It looks to me as if this will always return `null`. Please write a test for it.